### PR TITLE
Provide the Package Name in Semantic API

### LINKED ISSUE #27728: [Semantic API] Need package name of a symbol
**Description:**
Currently getting the symbol through Semantic API allows us to get the module name, version, orgName. Going forward we will need to get the package name for API Docs.

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by sanjana**
Fixes #27728

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/ballerinalang/model/elements/PackageID.java [hunk 1]
```diff
@@ -133,14 +135,28 @@ public PackageID(Name orgName, List<Name> nameComps, Name version) {
                 nameComps.stream()
                         .map(Name::getValue)
                         .collect(Collectors.joining(".")));
+        // TODO: The package name should be distinguishable when the pkgName != moduleName
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/org/ballerinalang/model/elements/PackageID.java:138]: Let's create an issue as well to track this.

Comment (sanjana) [reply compiler/ballerina-lang/src/main/java/org/ballerinalang/model/elements/PackageID.java:138]: Created a task issue at: #31154

**Comment by azinneera** [comment]
Project API chages LGTM

**Review by azinneera — COMMENTED** [review]
Project API changes LGTM

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionModuleId.java [hunk 2]
```diff
@@ -67,6 +67,17 @@ public String orgName() {
...
 
+    /**
+     * Get the Package name of this module ID.
+     *
+     * @return Package name
+     */
+    @Override
+    public String packageName() {
+        // TODO: Implement this method.
+        throw new UnsupportedOperationException("Not yet implemented");
```
Comment (pubudu91) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionModuleId.java:78]: @IMS94 Please note

Comment (pubudu91) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionModuleId.java:78]: Shall we create an issue to track this as well?

**Review by pubudu91 — APPROVED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionModuleId.java (hunk 2 above)
Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionModuleId.java:78]: Created https://github.com/ballerina-platform/ballerina-lang/issues/31459 to track this

**Pull request merged**